python/hail/methods/statgen.py

```diff
@@ -1454,16 +1451,17 @@ def pca(entry_expr, k=10, compute_loadings=False, as_array=False) -> Tuple[List[
     (:obj:`list` of :obj:`float`, :class:`.Table`, :class:`.Table`)
         List of eigenvalues, table with column scores, table with row loadings.
     """
-    source = entry_expr._indices.source
-    if not isinstance(source, MatrixTable):
-        raise ValueError("Expect an expression of 'MatrixTable', found {}".format(
-            "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
-    dataset = source
-    base, _ = dataset._process_joins(entry_expr)
-    analyze('pca', entry_expr, dataset._entry_indices)
-
-    r = Env.hail().methods.PCA.apply(dataset._jvds, to_expr(entry_expr)._ast.to_hql(), k, compute_loadings, as_array)
-    scores = Table(Env.hail().methods.PCA.scoresTable(dataset._jvds, as_array, r._2()))
+    mt = matrix_table_source('pca/entry_expr', entry_expr)
+
+    if entry_expr in mt._fields_inverse:
```

Comment:
I don't see a super compelling performance reason for this. Compared to the SVD itself, an extra copy should be basically free. Not worth the added complexity (and semantics change, see above) IMO.

Reply:
Isn't the concern more about needlessly invoking the compiler?

Reply:
at some point soon a `select_entries(__x = mt.x)` will be pretty much free.

Reply:
and if it's free, there's no reason to introduce extra complexity in the python